esphome/components/ili9xxx/display.py

```diff
@@ -77,20 +105,33 @@ def _validate(config):
         "TFT_2.4R",
         "ILI9341",
         "ILI9342",
+        "ST7789V",
     ]:
         raise cv.Invalid(
             "Provided model can't run on ESP8266. Use an ESP32 with PSRAM onboard"
         )
     return config
 
 
+PANEL_SCHEMA = cv.Schema(
+    {
+        cv.Optional(CONF_SWAP_XY, default=False): cv.boolean,
+        cv.Optional(CONF_MIRROR_X, default=False): cv.boolean,
+        cv.Optional(CONF_MIRROR_Y, default=False): cv.boolean,
+        cv.Optional(CONF_COLOR_ORDER, default="BGR"): cv.one_of(
+            *COLOR_ORDERS.keys(), upper=True
+        ),
+    }
+)
 CONFIG_SCHEMA = cv.All(
     font.validate_pillow_installed,
     display.FULL_DISPLAY_SCHEMA.extend(
         {
             cv.GenerateID(): cv.declare_id(ili9XXXSPI),
             cv.Required(CONF_MODEL): cv.enum(MODELS, upper=True, space="_"),
             cv.Optional(CONF_DIMENSIONS): cv.dimensions,
+            cv.Optional(CONF_WIDTH): cv.int_,
+            cv.Optional(CONF_HEIGHT): cv.int_,
```

Comment:
This should be like this
```suggestion
            cv.Optional(CONF_DIMENSIONS): cv.Any(
                cv.dimensions,
                cv.Schema({
                    cv.Optional(CONF_WIDTH): cv.int_,
                    cv.Optional(CONF_HEIGHT): cv.int_,
                }),
            ),
```

Reply:
Fixed in next push.